cvat/apps/engine/tests/utils.py

```diff
@@ -533,3 +556,228 @@ def filter_dict(
     d: dict[str, Any], *, keep: Sequence[str] = None, drop: Sequence[str] = None
 ) -> dict[str, Any]:
     return {k: v for k, v in d.items() if (not keep or k in keep) and (not drop or k not in drop)}
+
+
+def check_optional_fields(
+    self: TestCase,
+    obj: dict[str, int | float | str],
+    optional_values: dict[str, int | float | str],
+) -> None | NoReturn:
+    for k in optional_values.keys():
+        right = obj.get(k)
+        left = right if right is not None else optional_values[k]  # accounting for bool fields
+        if right is None:
+            right = optional_values[k]
+        self.assertEqual(
+            left,
+            right,
+            f"key {k!s} has incorrect actual optional value {left!s} instead of expected {right!s}",
+        )
```

Comment:
- could you rename left and right to something more self-explanatory? The values seem to be `actual` and `expected`, not simply elements of 2 objects.
- I can't find where the optional values are compared between 2 annotations
- consider using `{}.get(key, default)`